docs/v1-proposed-api.md

```diff
@@ -0,0 +1,226 @@
+# Proposed Changes for Hive V1 API
+
+Minimal changes, no new CRDs or relationships between.
+
+## ClusterDeployment
+
+ 1. Require an InstallConfig secret, referenced in a new ClusterDeployment.Spec field, to pass directly to installer.
+   * This will allow immediate use of new openshift-install features without needing to mirror in Hive's API.
+   * Remaining items in ClusterDeployment.Spec should be assumed mutable to every extent possible.
+   * Referenced as Spec.InstallConfigSecret. Secrets used as this config file contains ssh keys and pull secrets.
+   * Hive will parse the InstallConfig and copy relevant information back to the ClusterDeployment.
+     * What should happen if a field we support copyback for is set in the ClusterDeployment as well?
```

Comment:
I'm still not feeling right about this copyback functionality. Are there other examples of resources in Kube where you create resources A and B, and bits of B  get copied into A? I think I would prefer either duping the information that both installer and Hive need, or secondly Hive injecting parts into the InstallConfig. Will raise this on the call this morning.

Reply:
Early versions of the cluster-api v1alpha2 design called for status to be merged/copied from provider-specific CRs into the core CRs like Machine. I don't know if they actually implemented it that way. That's also only going in 1 direction, and it sounds like you're concerned about bidirectional copy/merge being confusing. I agree with that. I would only move data in one direction.

Reply:
AFAIU from installconfig to clusterprovision to cluster deployment we are copying data in one direction. Are you talking about the config management part on day 2?